proguard/src/main/java/com/flurry/proguard/UploadProguardMapping.java

```diff
@@ -11,184 +16,264 @@
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
 import java.net.HttpURLConnection;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import java.util.zip.GZIPOutputStream;
 
 public class UploadProguardMapping {
 
     private static final String METADATA_BASE = "https://crash-metadata.flurry.com/pulse/v1/";
     private static final String UPLOAD_BASE = "https://upload.flurry.com/upload/v1/upload/";
+    private static final int MAX_DURATION_SECONDS_DEFAULT = 600; // 10 Minutes
+    private static Logger LOGGER;
     private static HttpClient httpClient;
 
-    public static void main(String[] args) throws IOException, InterruptedException {
-        if (args.length != 3) {
-            System.out.println("Please provide 3 arguments in the following order:");
-            System.out.println("ApiKey PathToProguard AccessToken");
-            throw new IOException("Invalid number of arguments");
+    public static void main(String[] args) {
+
+        ArgumentParser parser = ArgumentParsers.newArgumentParser("com.flurry.proguard.UploadProguardMapping", true)
+                .description("Uploads Proguard Mapping Files for Android");
+        parser.addArgument("-k", "--api-key").help("API Key for your project").required(true);
+        parser.addArgument("-u", "--uuid").help("UUID").required(true);
+        parser.addArgument("-p", "--path").help("Path to Proguard file to be uploaded").required(true);
+        parser.addArgument("-t", "--token").help("Provide a valid Access Token").required(true);
+        parser.addArgument("-to", "--timeout").type(Integer.class).help(
+                "Provide a timeout value for Upload.")
+                .setDefault(MAX_DURATION_SECONDS_DEFAULT);
+
+        Namespace res = null;
+        try {
+            res = parser.parseArgs(args);
+        } catch (ArgumentParserException e) {
+            parser.handleError(e);
+            System.exit(1);
         }
+
         httpClient = HttpClientBuilder.create().build();
-        new UploadProguardMapping().uploadFile(args[0], args[1], args[2]);
+
+        Logger logger = Logger.getLogger(UploadProguardMapping.class.getName());
+        new UploadProguardMapping().uploadFile(res.getString("api_key"), res.getString("uuid"),
+                res.getString("path"), res.getString("token"), res.getInt("timeout"), logger);
     }
 
-    public void uploadFile(String apiKey, String pathToFiles, String token)
-            throws IOException, InterruptedException {
-        File file = new File(pathToFiles);
-        File zippedFile = tarZipFile(file);
+    public void uploadFile(String apiKey, String uuid, String pathToFile, String token, int timeout, Logger logger) {
+        LOGGER = logger;
```

Comment:
Changing to have 2 definitions for uploadFile, one with Logger param and the other without